### PR TITLE
feat: Add http endpoint to the Go feature server

### PR DESCRIPTION
**Pull request by tsotnet**
Signed-off-by: Tsotne Tabidze <tsotne@tecton.ai>

**What this PR does / why we need it**: This PR adds an HTTP endpoint to the Go feature server. We also add a flag to the CLI that will allow us to launch either HTTP or gRPC server.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: go/embedded/online_features.go [hunk 1]
```diff
@@ -243,6 +242,25 @@ func (s *OnlineFeatureService) StartGprcServer(host string, port int) error {
...
+		err := ser.Stop()
+		if err != nil {
+			fmt.Printf("Error when stopping the HTTP server: %v\n", err)
+		}
+	}()
+
+	return ser.Serve(host, port)
+}
+
 func (s *OnlineFeatureService) Stop() {
```
Comment (pyalex) [inline go/embedded/online_features.go:264]: Shouldn't we have Stop for http server as well?

File: go/internal/feast/server/http_server.go [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+	int64Val      []int64
+	doubleVal     []float64
+	boolVal       []bool
+	stringListVal [][]string
+	int64ListVal  [][]int64
+	doubleListVal [][]float64
+	boolListVal   [][]bool
+}
+
+func (u *repeatedValue) UnmarshalJSON(data []byte) error {
```
Comment (pyalex) [inline go/internal/feast/server/http_server.go:32]: Would be great to have some unit tests on this function

File: sdk/python/feast/embedded_go/online_features_service.py [hunk 3]
```diff
@@ -132,9 +132,15 @@ def get_online_features(
         resp = record_batch_to_online_response(record_batch)
         return OnlineResponse(resp)
 
+    def start_http_server(self, host: str, port: int):
```
Comment (pyalex) [inline sdk/python/feast/embedded_go/online_features_service.py:135]: We need at least some simple e2e test for this.

File: go/embedded/online_features.go [hunk 4]
```diff
@@ -243,6 +242,25 @@ func (s *OnlineFeatureService) StartGprcServer(host string, port int) error {
...
 
+func (s *OnlineFeatureService) StartHttpServer(host string, port int) error {
+	// TODO(oleksii): enable logging
+	var loggingService *logging.LoggingService = nil
+	ser := server.NewHttpServer(s.fs, loggingService)
+	log.Printf("Starting a HTTP server on host %s port %d\n", host, port)
+
+	go func() {
+		// As soon as these signals are received from OS, try to gracefully stop the gRPC server
+		<-s.grpcStopCh
```
[the remark this answers is not in the document]
Comment (tsotnet) [reply go/embedded/online_features.go]: I renamed this to `serverStopCh`. I could split it into 2 channels, but generally, you'd want to stop all running servers so I think we can leave it as is for now.

File: go/embedded/online_features.go (hunk 1 above)
Comment (tsotnet) [reply go/embedded/online_features.go:338]: This channel is common for both http & grpc (renamed variable for clarity)

File: sdk/python/feast/embedded_go/online_features_service.py (hunk 3 above)
Comment (tsotnet) [reply sdk/python/feast/embedded_go/online_features_service.py]: Added

File: go/internal/feast/server/http_server.go [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+	entitiesProto := make(map[string]*prototypes.RepeatedValue)
+	for key, value := range request.Entities {
+		entitiesProto[key] = value.ToProto()
+	}
+	requestContextProto := make(map[string]*prototypes.RepeatedValue)
+	for key, value := range request.RequestContext {
+		requestContextProto[key] = value.ToProto()
+	}
+
+	fmt.Printf("features: %+v\n", request.Features)
```
[the remark this answers is not in the document]
Comment (tsotnet) [reply go/internal/feast/server/http_server.go]: Removed

File: go/internal/feast/server/http_server.go (hunk 2 above)
Comment (tsotnet) [reply go/internal/feast/server/http_server.go:35]: Added

**Review by pyalex — APPROVED** [review]

**Pull request merged**